### PR TITLE
refactor: Remove unused SER_DISK, SER_NETWORK, CDataStream

### PR DESCRIPTION
**Pull request by maflcko**
Seems odd to have code that is completely dead.

Fix this by removing all of it.

### CODE REVIEW

File: src/hash.h [hunk 1]
```diff
@@ -223,15 +221,6 @@ class HashedSourceWriter : public HashWriter
     }
 };
 
-/** Compute the 256-bit hash of an object's serialization. */
-template<typename T>
-uint256 SerializeHash(const T& obj, int nType=SER_GETHASH, int nVersion=PROTOCOL_VERSION)
```
Comment (ajtowns) [inline src/hash.h:228]: Why get rid of this helper function? `SerializeHash(obj)` seems simpler to read than `(HashWriter{} << obj).GetHash()`.

Comment (ajtowns) [reply src/hash.h:228]: I guess most of the code is combining a few objects and hashing them, so I guess you're right...

Comment (ajtowns) [reply src/hash.h:228]:
```c++
    template <typename... Ts>
    explicit HashWriter(const Ts&... obj)
    {
        (*this << ... << obj);
    }
```

lets you replace

```c++
    uint64_t hash1 = (HashWriter{} << nKey << GetKey()).GetCheapHash();
    uint64_t hash2 = (HashWriter{} << nKey << netgroupman.GetGroup(*this) << (hash1 % ADDRMAN_TRIED_BUCKETS_PER_GROUP)).GetCheapHash();
```

with

```c++
    uint64_t hash1 = HashWriter{nKey, GetKey()}.GetCheapHash();
    uint64_t hash2 = HashWriter{nKey, netgroupman.GetGroup(*this), (hash1 % ADDRMAN_TRIED_BUCKETS_PER_GROUP)}.GetCheapHash();
```

Not sure it's worth it though.

**Review by ryanofsky — APPROVED** [review]
Seems odd to not code review ACK fa98a097a30bc39f2424c0efd28a7979155faae6 (looks good)

**Comment by ajtowns** [comment]
ACK fa98a097a30bc39f2424c0efd28a7979155faae6

**Comment by theuni** [comment]
> rebased and added a commit to rename version.h to node/protocol_version.h

Nice, thanks, I was going to PR the same thing :)

**Pull request merged**